middleware/request_id.go

```diff
@@ -0,0 +1,31 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/pborman/uuid"
+)
+
+// TODO
+// In Go, the idiomatic and recommended way to attach any request scoped
```

Comment:
I've thought of doing this several times. But each time attempt it, I just get confused at how `context` is supposed to be actually used, and don't proceed forward. But this is definitely something we need to do.